src/postamble.js

```diff
@@ -409,16 +409,14 @@ function exit(status, implicit) {
       var msg = 'program exited (with status: ' + status + '), but EXIT_RUNTIME is not set, so halting execution but not exiting the runtime or preventing further async execution (build with EXIT_RUNTIME=1, if you want a true shutdown)';
 #if MODULARIZE
       readyPromiseReject(msg);
-#else
-      err(msg);
 #endif // MODULARIZE
+      err(msg);
 #else
       var msg = 'program exited (with status: ' + status + '), but noExitRuntime is set due to an async operation, so halting execution but not exiting the runtime or preventing further async execution (you can use emscripten_force_exit, if you want to force a true shutdown)';
 #if MODULARIZE
       readyPromiseReject(msg);
-#else
-      err(msg);
 #endif // MODULARIZE
```

Comment:
Is this change logically separate or intertwined with the preamble changes?   Might be nice to separate it if possible?